pkg/webhook/gitmanager/init.go

```diff
@@ -39,14 +50,263 @@ type CustomRepo struct {
+			klog.ErrorS(err, "unable to retrieve the current working directory")
+			return nil, err
+		}
+		if path != "/" {
+			cr.Dir = path
+		}
+	}
+	cr.Dir += "/resource-auditing-repo"
+	r, err := git.PlainInit(cr.Dir, false)
+	if err == git.ErrRepositoryAlreadyExists {
+		klog.V(2).InfoS("network policy repository already exists - skipping initialization")
+		r, err := git.PlainOpen(cr.Dir)
+		if err != nil {
+			klog.ErrorS(err, "unable to retrieve existing repository")
+			return nil, err
+		}
+		return r, git.ErrRepositoryAlreadyExists
+	} else if err != nil {
+		klog.ErrorS(err, "unable to initialize git repo")
+		return nil, err
+	}
+	return r, nil
+}
+
+func (cr *CustomRepo) addResources() error {
+	if cr.StorageMode == StorageModeDisk {
+		os.Mkdir(cr.Dir+"/k8s-policies", 0700)
+		os.Mkdir(cr.Dir+"/antrea-policies", 0700)
+		os.Mkdir(cr.Dir+"/antrea-cluster-policies", 0700)
+		os.Mkdir(cr.Dir+"/antrea-tiers", 0700)
+	} else {
+		cr.Fs.MkdirAll("k8s-policies", 0700)
+		cr.Fs.MkdirAll("antrea-policies", 0700)
+		cr.Fs.MkdirAll("antrea-cluster-policies", 0700)
+		cr.Fs.MkdirAll("antrea-tiers", 0700)
+	}
+	if err := cr.addK8sPolicies(); err != nil {
+		klog.ErrorS(err, "unable to add K8s network policies to repository")
+		return err
+	}
+	if err := cr.addAntreaPolicies(); err != nil {
+		klog.ErrorS(err, "unable to add Antrea network policies to repository")
+		return err
+	}
+	if err := cr.addAntreaClusterPolicies(); err != nil {
+		klog.ErrorS(err, "unable to add Antrea cluster network policies to repository")
+		return err
+	}
+	if err := cr.addAntreaTiers(); err != nil {
+		klog.ErrorS(err, "unable to add Antrea tiers to repository")
+		return err
+	}
+	return nil
+}
+
+func (cr *CustomRepo) addK8sPolicies() error {
+	list := &unstructured.UnstructuredList{}
+	list.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "networking.k8s.io",
+		Version: "v1",
+		Kind:    "NetworkPolicyList",
+	})
+	policies, err := cr.K8s.ListResource(list)
+	if err != nil {
+		return err
+	}
+	var namespaces []string
+	for i, np := range policies.Items {
+		clearFields(&policies.Items[i])
+		name := np.GetName()
+		namespace := np.GetNamespace()
+		if !stringInSlice(namespace, namespaces) {
+			namespaces = append(namespaces, namespace)
+			if cr.StorageMode == StorageModeDisk {
+				os.Mkdir(cr.Dir+"/k8s-policies/"+namespace, 0700)
+			} else {
+				cr.Fs.MkdirAll("k8s-policies/"+namespace, 0700)
+			}
+		}
+		path := computePath(cr.Dir, "k8s-policies", namespace, name+".yaml")
+		y, err := yaml.Marshal(&policies.Items[i])
+		if err != nil {
+			klog.ErrorS(err, "unable to marshal policy config")
+			return err
+		}
+		if err := cr.writeFileToPath(path, y); err != nil {
+			klog.Errorf("unable to write k8s policy yaml to path %s", path)
+			return err
+		}
+		klog.V(2).Infof("Added K8s policy at resource-auditing-repo/k8s-policies/" + namespace + "/" + name + ".yaml")
+	}
+	return nil
+}
+
+func (cr *CustomRepo) addAntreaPolicies() error {
```

Comment:
I was hoping that using the unstructured client would avoid that code duplication. It feels to me we could have a single function with the `GroupVersionKind` as a parameter.

Reply:
Yes that should work - I'll go ahead and implement it that way now